lib/beaker/host.rb

```diff
@@ -224,13 +225,29 @@ def is_x86_64?
     end
 
     def connection
-      @connection ||= SshConnection.connect( reachable_name,
+      # create new connection object if necessary
+      @connection ||= SshConnection.connect( { :ip => self['ip'], :vmhostname => self['vmhostname'], :hostname => @name },
                                              self['user'],
                                              self['ssh'], { :logger => @logger } )
+      # update connection information
+      if self['ip'] && (@connection.ip != self['ip'])
```

Comment:
It was to make writing the spec tests easier - since you ask for the connection object every time that you do a host.exec it goes through this code path a lot and will end up resetting the value again and again.  Probably not the prettiest, but you also don't know if the ip has become available or if it is a new ip.   ip used to be considered constant but now we allow for change during runtime (at least, we've had situations where it has lost its ip lease and gotten a new one).